components/compliance-service/reporting/relaxting/reports.go

```diff
@@ -609,92 +648,90 @@ func (backend ES2Backend) getFiltersQuery(filters map[string][]string, latestOnl
 		boolQuery = boolQuery.Must(termQuery)
-	}
-
-	if len(filters["policy_name"]) > 0 {
-		termQuery := elastic.NewTermsQuery("policy_name", stringArrayToInterfaceArray(filters["policy_name"])...)
+	} else if latestOnly {
+		//only if there is no job_id filter set, do we want the daily latest
+		termQuery := elastic.NewTermsQuery("daily_latest", true)
 		boolQuery = boolQuery.Must(termQuery)
 	}
 
-	if len(filters["status"]) > 0 {
-		termQuery := elastic.NewTermsQuery("status", stringArrayToInterfaceArray(filters["status"])...)
-		boolQuery = boolQuery.Must(termQuery)
-	}
+	return boolQuery
+}
 
-	if len(filters["node_id"]) > 0 {
-		termQuery := elastic.NewTermsQuery("node_uuid", stringArrayToInterfaceArray(filters["node_id"])...)
-		boolQuery = boolQuery.Must(termQuery)
+func (backend ES2Backend) getESFieldName(filterType string) string {
+	switch filterType {
+	case "organization":
+		return "organization_name"
+	case "chef_server":
+		return "source_fqdn"
+	case "platform":
+		return "platform.name"
+	case "role":
+		return "roles"
+	case "recipe":
+		return "recipes"
+	case "inspec_version":
+		return "version"
+	default:
+		return filterType
 	}
+}
 
-	if len(filters["node_name"]) > 0 {
-		termQuery := elastic.NewTermsQuery("node_name", stringArrayToInterfaceArray(filters["node_name"])...)
-		boolQuery = boolQuery.Must(termQuery)
+func (backend ES2Backend) newTermQueryFromFilter(ESField string,
+	filters []string) *elastic.BoolQuery {
+	refinedValues := make([]string, 0, 0)
+	filterQuery := elastic.NewBoolQuery()
+
+	for _, value := range filters {
+		if containsWildcardChar(value) {
+			wildQuery := elastic.NewWildcardQuery(ESField, value)
+			filterQuery = filterQuery.Should(wildQuery)
+		} else {
+			refinedValues = append(refinedValues, value)
+		}
 	}
-
-	if len(filters["platform"]) > 0 {
-		termQuery := elastic.NewTermsQuery("platform.name", stringArrayToInterfaceArray(filters["platform"])...)
-		boolQuery = boolQuery.Must(termQuery)
+	if len(refinedValues) > 0 {
+		termQuery := elastic.NewTermsQuery(ESField, stringArrayToInterfaceArray(refinedValues)...)
+		filterQuery = filterQuery.Should(termQuery)
 	}
+	return filterQuery
+}
 
-	numberOfProfiles := len(filters["profile_id"])
-	numberOfControls := len(filters["control"])
-	if numberOfProfiles > 0 || numberOfControls > 0 {
-		profileBaseFscIncludes := []string{"profiles.name", "profiles.sha256", "profiles.version"}
-		profileLevelFscIncludes := []string{"profiles.controls_sums", "profiles.status"}
-		controlLevelFscIncludes := []string{"profiles.controls.id", "profiles.controls.status", "profiles.controls.impact"}
-
-		profileAndControlQuery := getProfileAndControlQuery(filters, profileBaseFscIncludes, profileLevelFscIncludes, controlLevelFscIncludes)
-		boolQuery = boolQuery.Must(profileAndControlQuery)
+func (backend ES2Backend) newNestedTermQueryFromFilter(ESField string, ESFieldPath string,
```

Comment:
The same as the `newTermQueryFromFilter` function but for nested array values like controls and profiles.